cpp/src/command_classes/CentralScene.cpp

```diff
@@ -162,34 +181,48 @@ bool CentralScene::HandleMsg
 	if( CentralSceneCmd_Set == (CentralSceneCmd)_data[0] )
 	{
 		// Central Scene Set received so send notification
-		int32 when;
-		if( _data[2] == 0 )
-			when = 0;
-		else if( _data[2] <= 0x7F )
-			when = _data[2];
-		else if( _data[2] <= 0xFE )
-			when = 60 * _data[2];
-		else
-			when = 0;
-		Log::Write( LogLevel_Info, GetNodeId(), "Received Central Scene set from node %d: scene id=%d in %d seconds. Sending event notification.", GetNodeId(), _data[3], when);
-
-		if( ValueInt* value = static_cast<ValueInt*>( GetValue( _instance, _data[3] ) ) )
+		uint8 keyAttribute = _data[2];
+		uint8 sceneID = _data[3];
+		Log::Write( LogLevel_Info, GetNodeId(), "Received Central Scene set from node %d: scene id=%d with key Attribute %d. Sending event notification.", GetNodeId(), sceneID, keyAttribute);
+
+		int index = 1;
+		if      ( keyAttribute == 0) { index = CentralScene_SinglePress; }
+		else if ( keyAttribute == 1) { index = CentralScene_Released; }
+		else if ( keyAttribute == 2) { index = CentralScene_HeldDown; }
+		else if ( keyAttribute == 3) { index = CentralScene_DoublePress; }
+		else if ( keyAttribute == 4) { index = CentralScene_TriplePress; }
+		else if ( keyAttribute == 5) { index = CentralScene_Pressed4Times; }
+		else if ( keyAttribute == 6) { index = CentralScene_Pressed5Times;}
+
```

Comment:
With this, there is no way to differentiate between different between the different actions that might take place on a Scene Controller. (eg, some devices allow you to program different "events" based on a single/double/triple click)